clouddriver-dcos/src/main/groovy/com/netflix/spinnaker/clouddriver/dcos/deploy/util/id/DcosSpinnakerLbId.groovy

```diff
@@ -0,0 +1,83 @@
+package com.netflix.spinnaker.clouddriver.dcos.deploy.util.id
+
+import static com.google.common.base.Preconditions.checkArgument
+import static com.google.common.base.Strings.nullToEmpty
+
+/**
+ * Represents a hierarchical Spinnaker specific load balancer identifier for DCOS.
+ * Structure - /account/loadBalancerName
+ */
+class DcosSpinnakerLbId {
+    public final static def SAFE_NAME_SEPARATOR = "_"
+
+    private final def marathonPath
+
+    public DcosSpinnakerLbId(final String id) {
+        this.marathonPath = MarathonPathId.parse(nullToEmpty(id)).absolute()
+
+        checkArgument(this.marathonPath.size() == 2, "A part of the DCOS Spinnaker LB ID was missing.")
+    }
+
+    public DcosSpinnakerLbId(final String account, final String loadBalancerName) {
+        checkArgument(!nullToEmpty(account).trim().empty, "The account should not be null, empty, or blank.")
+        checkArgument(!nullToEmpty(loadBalancerName).trim().empty, "The loadBalancerName should not be null, empty, or blank.")
+
+        marathonPath = MarathonPathId.parse("/${account}/${loadBalancerName}")
+    }
+
+    public String getAccount() {
+        marathonPath.first().get()
+    }
+
+    public String getLoadBalancerName() {
+        marathonPath.last().get()
+    }
+
+    public String getUnsafeLoadBalancerGroup() {
```

Comment:
Really glad you introduced this class, since I was too lazy at the time.

I think we can get rid of unsafeLoadBalancerGroup since we won't need that (it's the same as the marathon path basically). Then, consider naming getSafeLoadBalancerGroup() to just getLoadBalancerHaproxyGroup() to more accurately describe it.